src/OmniSharp.Roslyn.CSharp/Services/Intellisense/CompletionItemExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Completion;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Intellisense
+{
+    internal static class CompletionItemExtensions
+    {
+        private static MethodInfo _getSymbolsAsync;
+
+        static CompletionItemExtensions()
+        {
+            var symbolCompletionItemType = typeof(CompletionItem).GetTypeInfo().Assembly.GetType("Microsoft.CodeAnalysis.Completion.Providers.SymbolCompletionItem");
+            _getSymbolsAsync = symbolCompletionItemType.GetMethod("GetSymbolsAsync", BindingFlags.Public | BindingFlags.Static);
+        }
+
+        public static async Task<IEnumerable<ISymbol>> GetCompletionSymbols(this CompletionItem completionItem, IEnumerable<ISymbol> recommendedSymbols, Document document)
+        {
+            // for SymbolCompletionProvider, use the logic of extracting information from recommended symbols
```

Comment:
unfortunately since this change https://github.com/dotnet/roslyn/pull/15688, the `SymbolCompletionProvider` doesn't store encoded symbols in the properties of the completion item anymore, so just calling `GetSymbols()` results in an empty array. Instead, the only thing I could think of was that they are fetched using `Recommender`. I am sure you'd know more background behind this change and if there is a better way to do it?